sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AlreadyExistException.scala

```diff
@@ -61,6 +61,11 @@ class TableAlreadyExistsException(errorClass: String, messageParameters: Map[Str
     this(errorClass = "TABLE_OR_VIEW_ALREADY_EXISTS",
       messageParameters = Map("relationName" -> quoteNameParts(table)))
   }
+
+  def this(tableIdent: Identifier) = {
```

Comment:
Nit: let's also add a one-line comment saying this is for backward compatibility and it is used by external connectors. Otherwise, developers may remove it again after finding it is not used in Spark.

Reply:
Added to each class used in the connector API as sometimes I had to add back multiple constructors.